File: testdata/vulnerable_crypto.go
package testdata

import (
	"crypto/des"
	"crypto/ecdsa"
	"crypto/elliptic"
	"crypto/md5"
	"crypto/rand"
	"crypto/rc4"
	"crypto/rsa"
	"crypto/sha1"
)

// RSA key generation — quantum-vulnerable
func genRSAKey() {
	rsa.GenerateKey(rand.Reader, 1024)
}

// RSA encryption — quantum-vulnerable
func encryptRSA(pub *rsa.PublicKey, data []byte) {
	rsa.EncryptPKCS1v15(rand.Reader, pub, data)
}

// RSA signing — quantum-vulnerable
func signRSA(priv *rsa.PrivateKey, digest []byte) {
	rsa.SignPKCS1v15(rand.Reader, priv, 0, digest)
}

// ECDSA key generation — quantum-vulnerable
func genECDSA() {
	ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
}

// ECDSA signing — quantum-vulnerable
func signECDSA(priv *ecdsa.PrivateKey, digest []byte) {
	ecdsa.SignASN1(rand.Reader, priv, digest)
}

// 3DES — deprecated
func tripleDES(key []byte) {
	des.NewTripleDESCipher(key)
}

// DES — deprecated
func desCipher(key []byte) {
	des.NewCipher(key)
}

// RC4 — broken
func rc4Cipher(key []byte) {
	rc4.NewCipher(key)
}

// MD5 — collision-broken
func hashMD5(data []byte) {
	h := md5.New()
	h.Write(data)
	md5.Sum(data)
}

// SHA-1 — deprecated
func hashSHA1(data []byte) {
	h := sha1.New()
	h.Write(data)
	sha1.Sum(data)
}

File: testdata/VulnerableCrypto.java
import java.security.*;
import java.security.spec.RSAKeyGenParameterSpec;
import javax.crypto.*;
import java.math.BigInteger;

public class VulnerableCrypto {

    // RSA key generation — quantum-vulnerable
    public KeyPair generateRSAKey() throws Exception {
        KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
        kpg.initialize(new RSAKeyGenParameterSpec(1024, BigInteger.valueOf(65537)));
        return kpg.generateKeyPair();
    }

    // RSA encryption
    public byte[] encryptRSA(PublicKey pub, byte[] data) throws Exception {
        Cipher cipher = Cipher.getInstance("RSA/ECB/PKCS1Padding");
        cipher.init(Cipher.ENCRYPT_MODE, pub);
        return cipher.doFinal(data);
    }

    // ECDH key agreement — quantum-vulnerable
    public byte[] ecdhKeyAgreement(PrivateKey priv, PublicKey pub) throws Exception {
        KeyAgreement ka = KeyAgreement.getInstance("ECDH");
        ka.init(priv);
        ka.doPhase(pub, true);
        return ka.generateSecret();
    }

    // ECDSA signing — quantum-vulnerable
    public byte[] ecdsaSign(PrivateKey priv, byte[] data) throws Exception {
        Signature sig = Signature.getInstance("SHA256withECDSA");
        sig.initSign(priv);
        sig.update(data);
        return sig.sign();
    }

    // AES in ECB mode — insecure mode
    public byte[] encryptAESECB(SecretKey key, byte[] data) throws Exception {
        Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
        cipher.init(Cipher.ENCRYPT_MODE, key);
        return cipher.doFinal(data);
    }

    // AES with no mode — defaults to ECB on most JVMs
    public byte[] encryptAESDefault(SecretKey key, byte[] data) throws Exception {
        Cipher cipher = Cipher.getInstance("AES");
        cipher.init(Cipher.ENCRYPT_MODE, key);
        return cipher.doFinal(data);
    }

    // Triple DES — deprecated
    public byte[] encryptTripleDES(SecretKey key, byte[] data) throws Exception {
        Cipher cipher = Cipher.getInstance("DESede/CBC/PKCS5Padding");
        cipher.init(Cipher.ENCRYPT_MODE, key);
        return cipher.doFinal(data);
    }

    // RC4 — broken stream cipher
    public byte[] encryptRC4(SecretKey key, byte[] data) throws Exception {
        Cipher cipher = Cipher.getInstance("RC4");
        cipher.init(Cipher.ENCRYPT_MODE, key);
        return cipher.doFinal(data);
    }

    // MD5 — collision-broken
    public byte[] hashMD5(byte[] data) throws Exception {
        MessageDigest md = MessageDigest.getInstance("MD5");
        return md.digest(data);
    }

    // SHA-1 — deprecated
    public byte[] hashSHA1(byte[] data) throws Exception {
        MessageDigest md = MessageDigest.getInstance("SHA-1");
        return md.digest(data);
    }

    // SHA1withRSA — double vulnerability
    public byte[] signSHA1withRSA(PrivateKey priv, byte[] data) throws Exception {
        Signature sig = Signature.getInstance("SHA1withRSA");
        sig.initSign(priv);
        sig.update(data);
        return sig.sign();
    }
}
